### ISSUE TITLE
Enhancement: support Node.js builtin modules in `readonly-parameter-types`

### ISSUE THREAD

**Issue by arka1002** [typescript-eslint/typescript-eslint#8572]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

type-utils

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

A few lines of context - 
1. An old rfc - https://github.com/typescript-eslint/typescript-eslint/discussions/6017#discussion-4576461
2. StyleShit's comment on a PR on which I'm working - https://github.com/typescript-eslint/typescript-eslint/pull/8502#discussion_r1494653089

Steps to reproduce -
1. go to https://stackblitz.com/edit/stackblitz-starters-tnrbtd?file=src%2Findex.ts
2. run `npx eslint src/index.ts` in terminal
3. check index.ts, line 4

Is typescript-eslint open to include a config option like `{from: "package", name: "Foo", package: "data:text/javascript, export class Foo {};"}`, or with the other specifiers(like `file:`), in the eslint config file ?

**Comment by arka1002** [comment]
Thanks for the quick answer, so its only `node:` which will be supported ?

**Comment by bradzacher** [comment]
This isn't an enhancement - it's a bug.

We don't handle the node scope packages in this function
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages%2Ftype-utils%2Fsrc%2FTypeOrValueSpecifier.ts#L155-L172

Which means we're looking for `@types/node:events`.
I'd guess a lot of the node internal packages might fail that lookup given that `events` maps to `@types/node`?

**Comment by arka1002** [comment]
> We don't handle the node special scoped packages in this function

Yes, I also tried to solve it [here](https://github.com/typescript-eslint/typescript-eslint/pull/8502#discussion_r1493811778) in a rough way.

So, the only URL scheme which is allowed, is `node:` right ?

**Comment by auvred** [comment]
What about `bun:` prefixed packages? https://bun.sh/docs/runtime/bun-apis

**Comment by bradzacher** [comment]
Whilst technically they are URL schemes I don't think we should be thinking about then like that.

Instead we should just be treating them as a special namespace case. Eg `node:foo` maps to `@types/node` because we see the colon, split it, and take the first part as the namespace to use.

**Comment by Josh-Cena** [comment]
> As for `data:text/JavaScript` - what does that even mean? That you would just execute random JS? It's honestly non-sensical when the option is looking for an npm package name.

Here's a trivia:

```js
import { Foo } from "data:text/javascript, export class Foo {};"

console.log(new Foo());
```

This is valid code :)
